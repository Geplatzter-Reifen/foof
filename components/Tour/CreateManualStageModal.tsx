import React, { useState } from "react";
import { Button, Card, Input, Modal, Text } from "@ui-kitten/components";
import { createManualStage } from "@/services/tracking";
import { StyleSheet } from "react-native";

interface CreateManualStageModalProps {
  isVisible: boolean;
  onClose: () => void;
  tourId: string;
}

export const CreateManualStageModal: React.FC<CreateManualStageModalProps> = ({
  isVisible,
  onClose,
  tourId,
}) => {
  const [stageName, setStageName] = useState("");
  const [startCoords, setStartCoords] = useState("");
  const [endCoords, setEndCoords] = useState("");
  const [errorText, setErrorText] = useState("");

  const reset = () => {
    onClose();
    setStageName("");
    setStartCoords("");
    setEndCoords("");
    setErrorText("");
  };

  const onSave = async () => {
    try {
      await createManualStage(stageName, startCoords, endCoords, tourId);
      reset();
    } catch (err) {
      if (err instanceof Error) {
        setErrorText(err.message);
      }
    }
  };

  return (
    <Modal visible={isVisible} backdropStyle={styles.backdrop}>
      <Card disabled={true} style={styles.dialog}>
        <Input
          placeholder="Tourname"
          value={stageName}
          onChangeText={(stageText) => setStageName(stageText)}
          style={styles.input}
        />
        <Input
          placeholder="Startkoordinaten"
          value={startCoords}
          onChangeText={(coordsText) => setStartCoords(coordsText)}
          style={styles.input}
        />
        <Input
          placeholder="Endkoordinaten"
          value={endCoords}
          onChangeText={(coordsText) => setEndCoords(coordsText)}
          style={styles.input}
        />
        {errorText && <Text>{errorText}</Text>}
        <Button onPress={onSave}>Speichern</Button>
        <Button status="basic" onPress={reset} style={styles.input}>
          Abbrechen
        </Button>
      </Card>
    </Modal>
  );
};

const styles = StyleSheet.create({
  backdrop: {
    width: "100%",
    backgroundColor: "rgba(0,0,0,0.5)",
  },
  dialog: {
    padding: 0,
    width: 360,
  },
  input: {
    marginBottom: 5,
  },
});
