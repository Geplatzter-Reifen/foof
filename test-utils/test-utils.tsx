// https://testing-library.com/docs/react-testing-library/setup/#add-custom-queries
import React from "react";
import { render, RenderOptions } from "@testing-library/react-native";
import { FontAwesomeIconsPack } from "@/components/Font/fontAwesome";
import { mapping } from "@eva-design/eva";
import {
  IconRegistry,
  ApplicationProvider,
  ModalService,
} from "@ui-kitten/components";
import * as eva from "@eva-design/eva";
import { foofDarkTheme, foofLightTheme } from "@/constants/custom-theme";
import { library } from "@fortawesome/fontawesome-svg-core";
import { far } from "@fortawesome/free-regular-svg-icons";
import { fas } from "@fortawesome/free-solid-svg-icons";
import { fab } from "@fortawesome/free-brands-svg-icons";

// Iconpacks laden
library.add(far, fas, fab);

// ModalService wird verwendet, um die Modal-Positionierung zu steuern
ModalService.setShouldUseTopInsets = true;

type Props = {
  children: React.ReactNode;
  use_dark_theme?: boolean;
};
// Der Wrapper wird verwendet, um die Provider zu umschließen, die für die Komponenten notwendig sind
const AllTheProviders = ({ children, use_dark_theme = true }: Props) => {
  const theme = use_dark_theme
    ? { ...eva.dark, ...foofDarkTheme }
    : { ...eva.light, ...foofLightTheme };
  return (
    <>
      <IconRegistry icons={FontAwesomeIconsPack} />
      <ApplicationProvider {...eva} customMapping={mapping} theme={theme}>
        {children}
      </ApplicationProvider>
    </>
  );
};

// Erstellt einen Custom-Render, der die Provider um die Komponenten wickelt
const customRender = (
  component: React.ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => {
  return render(component, { wrapper: AllTheProviders, ...options });
};

// Exportiert alle Funktionen der Testing Library, weswegen man diese dann auch von dieser Datei importieren kann
export * from "@testing-library/react-native";
export { customRender as render };
