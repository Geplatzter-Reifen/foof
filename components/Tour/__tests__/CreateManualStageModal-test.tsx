import { render } from "@/test-utils/test-utils";
import { CreateManualStageModal } from "../CreateManualStageModal";

// Ich vermute das dieser Test nicht mehr n√∂tig ist, da die Komponente nicht mehr existiert.
describe("<CreateManualStageModal />", () => {
  it("should render correctly", () => {
    const view = render(
      <CreateManualStageModal
        isVisible={true}
        onClose={jest.fn()}
        tourId="1"
      />,
    );
    expect(view).toMatchSnapshot();
  });
});
