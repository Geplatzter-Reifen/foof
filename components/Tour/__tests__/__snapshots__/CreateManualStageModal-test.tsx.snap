// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateManualStageModal /> should render correctly 1`] = `
<Modal
  hardwareAccelerated={false}
  statusBarTranslucent={false}
  transparent={true}
  visible={true}
>
  <View
    style={
      {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "rgba(0,0,0,0.5)",
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "width": "100%",
        }
      }
      testID="@backdrop"
    />
    <View
      backdropStyle={
        {
          "backgroundColor": "rgba(0,0,0,0.5)",
          "width": "100%",
        }
      }
      onLayout={[Function]}
      shouldUseContainer={true}
      style={
        [
          undefined,
          {
            "position": "absolute",
          },
          {
            "left": -999,
            "top": -999,
          },
        ]
      }
      visible={true}
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": true,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#373737",
            "borderColor": "#FFFFFF",
            "borderRadius": 4,
            "borderWidth": 1,
            "opacity": 1,
            "overflow": "hidden",
            "padding": 0,
            "width": 360,
          }
        }
      >
        <View
          style={
            {
              "backgroundColor": undefined,
              "height": 0,
            }
          }
        />
        <View
          style={
            [
              {
                "flexShrink": 1,
              },
              {
                "paddingHorizontal": 24,
                "paddingVertical": 16,
              },
            ]
          }
        >
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginBottom": 5,
                "opacity": 1,
              }
            }
            testID="@undefined/container"
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#282828",
                    "borderColor": "#FFFFFF",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "minHeight": 40,
                    "paddingHorizontal": 8,
                    "paddingVertical": 7,
                  },
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                ]
              }
            >
              <TextInput
                appearance="default"
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                placeholder="Tourname"
                placeholderTextColor="#282828"
                size="medium"
                status="basic"
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "normal",
                      "marginHorizontal": 8,
                    },
                    {
                      "flexBasis": "auto",
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                    null,
                    undefined,
                  ]
                }
                testID="@undefined/input"
                value=""
              />
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginBottom": 5,
                "opacity": 1,
              }
            }
            testID="@undefined/container"
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#282828",
                    "borderColor": "#FFFFFF",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "minHeight": 40,
                    "paddingHorizontal": 8,
                    "paddingVertical": 7,
                  },
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                ]
              }
            >
              <TextInput
                appearance="default"
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                placeholder="Startkoordinaten"
                placeholderTextColor="#282828"
                size="medium"
                status="basic"
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "normal",
                      "marginHorizontal": 8,
                    },
                    {
                      "flexBasis": "auto",
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                    null,
                    undefined,
                  ]
                }
                testID="@undefined/input"
                value=""
              />
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={false}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "marginBottom": 5,
                "opacity": 1,
              }
            }
            testID="@undefined/container"
          >
            <View
              style={
                [
                  {
                    "backgroundColor": "#282828",
                    "borderColor": "#FFFFFF",
                    "borderRadius": 4,
                    "borderWidth": 1,
                    "minHeight": 40,
                    "paddingHorizontal": 8,
                    "paddingVertical": 7,
                  },
                  {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "width": "100%",
                  },
                ]
              }
            >
              <TextInput
                appearance="default"
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
                placeholder="Endkoordinaten"
                placeholderTextColor="#282828"
                size="medium"
                status="basic"
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "fontFamily": "System",
                      "fontSize": 15,
                      "fontWeight": "normal",
                      "marginHorizontal": 8,
                    },
                    {
                      "flexBasis": "auto",
                      "flexGrow": 1,
                      "flexShrink": 1,
                    },
                    null,
                    undefined,
                  ]
                }
                testID="@undefined/input"
                value=""
              />
            </View>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#FF6E14",
                "borderColor": "#FF6E14",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 40,
                "minWidth": 40,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              appearance="default"
              category="p1"
              status="basic"
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  {
                    "color": "#000000",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "marginHorizontal": 10,
                  },
                ]
              }
            >
              Speichern
            </Text>
          </View>
          <View
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              {
                "alignItems": "center",
                "backgroundColor": "#171717",
                "borderColor": "#171717",
                "borderRadius": 4,
                "borderWidth": 1,
                "flexDirection": "row",
                "justifyContent": "center",
                "marginBottom": 5,
                "minHeight": 40,
                "minWidth": 40,
                "opacity": 1,
                "paddingHorizontal": 10,
                "paddingVertical": 12,
              }
            }
          >
            <Text
              appearance="default"
              category="p1"
              status="basic"
              style={
                [
                  {
                    "color": "#FFFFFF",
                    "fontFamily": "System",
                    "fontSize": 15,
                    "fontWeight": "400",
                  },
                  {
                    "color": "#7e7e7e",
                    "fontFamily": "System",
                    "fontSize": 14,
                    "fontWeight": "bold",
                    "marginHorizontal": 10,
                  },
                ]
              }
            >
              Abbrechen
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
